src/watcher.ts

```diff
@@ -84,12 +84,12 @@ export class RedisWatcher implements Watcher {
     return watcher;
   }
 
-  private constructor() {}
+  public constructor() {}
 
-  public async update(): Promise<boolean> {
+  public async update(msg?: string): Promise<boolean> {
     const client = await this.pubConnection.getRedisClient();
     // @ts-ignore - publish is exists.
-    await client.publish(this.channel, 'casbin rules updated');
+    await client.publish(this.channel, msg ? msg : 'casbin rules updated');
```

Comment:
Is this necessary? Maybe you can override it.

Reply:
@Zxilly I was wondering that I would have to re-write for override like the same Redis connections and all that, so I was thinking to update this.

Reply:
@nodece @Zxilly , what about we add watcherEx class inside this project only and then it depends on the user which one they want to use?

Reply:
@Shivansh-yadav13 and I talked about this offline. This PR is unnecessary, so close this PR.